core/templates/pages/skill-editor-page/skill-editor-page.component.spec.ts

```diff
@@ -361,6 +361,67 @@ describe('Skill editor page', () => {
     expect(routingSpy).toHaveBeenCalled();
   });
 
+  it(
+    'should show save pending changes modal and navigate to ' +
+      'main editor tab when user confirms',
+    fakeAsync(() => {
+      skillEditorRoutingService.questionIsBeingCreated = true;
+      let routingSpy = spyOn(skillEditorRoutingService, 'navigateToMainTab');
+      spyOn(ngbModal, 'open').and.callFake((dlg, opt) => {
+        return {
+          componentInstance: MockNgbModalRef,
+          result: Promise.resolve('confirm'),
+        } as NgbModalRef;
+      });
+      component.selectMainTab();
+      tick();
+      expect(ngbModal.open).toHaveBeenCalled();
+      expect(routingSpy).toHaveBeenCalled();
+      skillEditorRoutingService.questionIsBeingCreated = false;
+    })
+  );
+
+  it(
+    'should show save pending changes modal and navigate to ' +
+      'preview editor tab when user confirms',
+    fakeAsync(() => {
+      skillEditorRoutingService.questionIsBeingCreated = true;
+      let routingSpy = spyOn(skillEditorRoutingService, 'navigateToPreviewTab');
+      spyOn(ngbModal, 'open').and.callFake((dlg, opt) => {
+        return {
+          componentInstance: MockNgbModalRef,
+          result: Promise.resolve('confirm'),
+        } as NgbModalRef;
+      });
+      component.selectPreviewTab();
+      tick();
+      expect(ngbModal.open).toHaveBeenCalled();
+      expect(routingSpy).toHaveBeenCalled();
+      skillEditorRoutingService.questionIsBeingCreated = false;
+    })
+  );
+
+  it(
+    'should show save pending changes modal and should not navigate to ' +
+      'preview editor tab when user cancels',
+    fakeAsync(() => {
+      skillEditorRoutingService.questionIsBeingCreated = true;
+      spyOn(ngbModal, 'open').and.callFake((dlg, opt) => {
+        return {
+          componentInstance: MockNgbModalRef,
+          result: Promise.reject(),
+        } as NgbModalRef;
+      });
+      component.selectPreviewTab();
+      tick();
+      spyOn(skillEditorRoutingService, 'getActiveTabName').and.returnValue(
+        'questions'
+      );
+      expect(component.getActiveTabName()).toBe('questions');
```

Comment:
Hm, wouldn't this return "questions" regardless of the tab navigation because we are spying on `getActiveTabName`. 

Perhaps we should execute the actual method (i.e. not spyOn) before and after `component.selectPreviewTab()` and expect the result to be "questions".